Packages/com.unity.multiplayer.samples.coop/Utilities/SceneManagement/SceneLoader.cs

```diff
@@ -0,0 +1,137 @@
+using System.Collections.Generic;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Unity.Multiplayer.Samples.Utilities
+{
+    public class SceneLoader : NetworkBehaviour
+    {
+        [SerializeField]
+        List<string> m_ScenesTriggeringLoadingScreen;
+
+        [SerializeField]
+        List<string> m_ScenesEndingLoadingScreen;
```

Comment:
That's what I did originally, but it doesn't work as well when loading the BossRoom scene. In that case, it would stop the loading screen before the first additive scene loads, and before the player characters spawn, which is what it was supposed to hide. Basically, I need a way to prevent the loading screen from stopping when the BossRoom scene is loaded and have it wait for all additive scenes. But I agree it is not an elegant solution.

I could get the RTT and add it as a delay before stopping the loading screen, which should give enough time to start loading the additive scenes, but I'm not sure how to do this cleanly yet. Plus I'm not a fan of adding these kinds of delays, it feels like a workaround. I will try to think of something else.

Reply:
Nevermind, it works. The issue I remembered was probably due to me using the wrong events in the beginning of this PR.